### PR TITLE
Add SystemProperties dictionary to binding data

### LINKED ISSUE #2068: Cannot access IOT specific message properties in functions V2
In functions V1, a JS function with an eventhub trigger that was wired to an IOT hub would see a bunch of IOT specific metadata:

```
bindingData:
    …
    "systemPropertiesArray": [
      {
        "iothub-connection-device-id": "miye-device0",
        "iothub-connection-auth-method": "{\"scope\":\"device\",\"type\":\"sas\",\"issuer\":\"iothub\",\"acceptingIpFilterRule\":null}",
        "iothub-connection-auth-generation-id": "636640121350566756",
        "iothub-enqueuedtime": "2018-12-13T14:42:43.358Z",
        "iothub-message-source": "Telemetry",
        "x-opt-sequence-number": 6127,
        "x-opt-offset": "3015760",
        "x-opt-enqueued-time": "2018-12-13T14:42:43.403Z",
        "enqueuedTimeUtc": "2018-12-13T14:42:43.403Z",
        "sequenceNumber": 6127,
        "offset": "3015760"
      }
    ],
```

In V2, this metadata is not accessible:

```
bindingData:
   …
  "systemPropertiesArray": [
      {
        "SequenceNumber": 6128,
        "Offset": "3016216",
        "PartitionKey": null,
        "EnqueuedTimeUtc": "2018-12-13T14:55:43.94Z"
      }
    ],
   …
```

Some testing with the Event Hubs .NET Standard SDK indicates that this is not due to a change in event hubs SDK behavior and its most likely that these properties are somehow filtered out/hidden by functions. Further investigation is required.

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #2068

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Extensions/SystemPropertiesCollectionExtensions.cs [hunk 1]
```diff
@@ -11,13 +10,14 @@ static internal class SystemPropertiesCollectionExtensions
...
-            };
+            IDictionary<string, object> modifiedDictionary = collection;
+            
+            // Following is needed to maintain structure of bindingdata: https://github.com/Azure/azure-webjobs-sdk/pull/1849
+            modifiedDictionary["SequenceNumber"] = collection.SequenceNumber;
+            modifiedDictionary["Offset"] = collection.Offset;
+            modifiedDictionary["PartitionKey"] = collection.PartitionKey;
+            modifiedDictionary["EnqueuedTimeUtc"] = collection.EnqueuedTimeUtc;
+            return modifiedDictionary;
         }
```
Comment (paulbatum) [inline src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Extensions/SystemPropertiesCollectionExtensions.cs]: Is it right to be modifying the SystemPropertiesCollection that was passed in here? Feels like we should be making a copy of that collection, and then modifying the copy. Let me know if I'm missing something.

Comment (pragnagopa) [reply src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Extensions/SystemPropertiesCollectionExtensions.cs]: You are right. I did not intend to change the original collection. Will send follow up PR.

Comment (mathewc) [reply src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Extensions/SystemPropertiesCollectionExtensions.cs]: Preferable, I agree. That way we aren't modifying the instance we got from EventData.SystemProperties.

**Comment by jsaedtler** [comment]
I'm looking forward having this deployed and seeing the system properties in azure functions that are listening on event hubs that are populated by the iot hub routing.
When can I expect this to be online?

**Pull request merged**